source/extensions/filters/network/thrift_proxy/router/router_impl.h

```diff
@@ -180,15 +180,16 @@ class Router : public Tcp::ConnectionPool::UpstreamCallbacks,
          Stats::Scope& scope)
       : cluster_manager_(cluster_manager), stats_(generateStats(stat_prefix, scope)),
         stat_name_set_(scope.symbolTable().makeSet("thrift_proxy")),
-        request_call_(stat_name_set_->add("request_call")),
-        request_oneway_(stat_name_set_->add("request_oneway")),
-        request_invalid_type_(stat_name_set_->add("request_invalid_type")),
-        response_reply_(stat_name_set_->add("response_reply")),
-        response_reply_success_(stat_name_set_->add("response_success")),
-        response_reply_error_(stat_name_set_->add("response_error")),
-        response_exception_(stat_name_set_->add("response_exception")),
-        response_invalid_type_(stat_name_set_->add("response_invalid_type")),
-        passthrough_supported_(false) {}
+        symbol_table_(scope.symbolTable()), thrift_(stat_name_set_->add("thrift")),
+        upstream_rq_call_(stat_name_set_->add("upstream_rq_call")),
+        upstream_rq_oneway_(stat_name_set_->add("upstream_rq_oneway")),
+        upstream_rq_invalid_type_(stat_name_set_->add("upstream_rq_invalid_type")),
+        upstream_resp_reply_(stat_name_set_->add("upstream_resp_reply")),
+        upstream_resp_reply_success_(stat_name_set_->add("upstream_resp_success")),
+        upstream_resp_reply_error_(stat_name_set_->add("upstream_resp_error")),
+        upstream_resp_exception_(stat_name_set_->add("upstream_resp_exception")),
+        upstream_resp_invalid_type_(stat_name_set_->add("upstream_resp_invalid_type")),
+        upstream_rq_time_(stat_name_set_->add("upstream_rq_time")), passthrough_supported_(false) {}
```

Comment:
I wonder if we should just add the `thrift.` prefix to every one of these stats here, to avoid the runtime concat of these two stats... [ I see it done for some of the TLS code... ]

Reply:
I see both ways being done in the repo

Reply:
Ah ok, np then.